### PR TITLE
Change schemes to encodings

### PR DESCRIPTION
**Pull request by kriskowal**

### CODE REVIEW

File: encodings/json.js [hunk 1]
```diff
@@ -24,35 +24,35 @@ var Result = require('rezult');
 
 var EMPTY_OBJECT = {};
 
-module.exports = JsonScheme;
+module.exports = JsonEncoding;
```
Comment (breerly) [inline encodings/json.js]: I think we want the package name to be `encoding` not `encodings` - @abhinav has been trying to stomp out plurals awhile now, and now I agree with him.

File: examples/echo-app.js [hunk 2]
```diff
@@ -23,7 +23,7 @@
 var Rpc = require('../rpc');
 var HttpInbound = require('../transports/http/inbound');
```
Comment (breerly) [inline examples/echo-app.js]: Let's switch to singular for `transports` and `encodings`.

File: rpc.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ function Rpc(args) {
 Rpc.prototype.httpCallerHeader = 'YARPC-Source';
 Rpc.prototype.httpServiceHeader = 'YARPC-Service';
 Rpc.prototype.httpProcedureHeader = 'YARPC-Procedure';
-Rpc.prototype.httpSchemeHeader = 'YARPC-Scheme';
+Rpc.prototype.httpEncodingHeader = 'YARPC-Encoding';
```
Comment (breerly) [inline rpc.js]: Do we want `YARPC-*` over the wire or `Yarpc-*`?

File: test/http-json-outbound.js [hunk 4]
```diff
@@ -172,13 +172,13 @@ test('bad http handlers', function t(assert) {
                 callback();
             },
```
Comment (breerly) [inline test/http-json-outbound.js]: Are test file paths not supposed to map 1to1 to the file they are testing?

**Comment by breerly** [comment]
Mostly good - had some naming things that can probably be solved in another PR.

:shipit:

File: test/http-json-outbound.js (hunk 4 above)
Comment (kriskowal) [inline test/http-json-outbound.js]: There is no prevailing convention.

File: rpc.js (hunk 3 above)
Comment (kriskowal) [inline rpc.js]: cc @abhinav

File: examples/echo-app.js (hunk 2 above)
Comment (kriskowal) [inline examples/echo-app.js]: I’ll depluralize both in a follow-up.

**Pull request merged**